instrument/extended.go

```diff
@@ -21,26 +21,99 @@
 package instrument
 
 import (
+	"fmt"
 	"os"
 	"runtime"
+	"strings"
 	"sync/atomic"
 	"time"
 
-	"github.com/shirou/gopsutil/process"
+	"github.com/m3db/m3x/process"
+
 	"github.com/uber-go/tally"
 )
 
+// ExtendedMetricsType is a type of extended metrics to report.
+type ExtendedMetricsType int
+
+const (
```

Comment:
Perhaps we might want to use a bitflag to determine which metrics to report or not so we have more fine-grained control in the future? I can see scenarios where we're testing builds of the collector where we want to report a certain metric that is contained in `DetailedExtendedMetrics` (for example, GC pause times) but don't need the other metrics. In this case, we can still have constants for no extended metrics,  simple extended metrics, and detailed extended metrics which will be equal to their respective bitflag integers, and be used in the majority of cases to simplify things without sacrificing the option to turn individual metrics on or off.